nextcloudappstore/scaffolding/forms.py

```diff
@@ -86,29 +85,39 @@ class IntegrationScaffoldingForm(Form):
                                         ' integration '
                                         'does. Can contain Markdown.'))
 
-    def save(self, user, app_id):
+    def save(self, user, app_id, action):
         if app_id is None:
             app_id = self.cleaned_data['name'].lower().replace(" ", "_")
         try:
             app = App.objects.get(id=app_id)
-            if app.can_update(user):
-                '''Not optimal but works'''
-                Screenshot.objects.filter(app=app).delete()
-                if self.data['screenshot']:
-                    screenshot = Screenshot.objects.create(
-                        url=self.cleaned_data['screenshot'],
-                        small_thumbnail=self.cleaned_data[
-                            'screenshot_thumbnail'],
-                        ordering=1, app=app)
-                    screenshot.save()
+            if app.can_update(user) or user.is_superuser:
+                if action == "reject":
+                    Screenshot.objects.filter(app=app).delete()
+                    app.delete()
+                elif action == "approve":
+                    app.approved = True
+                    if settings.DISCOURSE_TOKEN:
+                        self._create_discourse_category(app_id)
+                    app.save()
+                    return app_id
```

Comment:
does it mean you as intgeration-developer can you self-approve?

Reply:
Uh, good catch - will check.